File: 2272-substring-with-largest-variance/2272-substring-with-largest-variance.py
class Solution:
    def largestVariance(self, s: str) -> int:
        ls = list(set(s))
        counts = Counter(s)
        ans = 0
        for x in range(len(ls)):
            a = ls[x]
            for y in range(len(ls)):
                if x == y:
                    continue
                b = ls[y]
                a_, b_ = 0, 0
                b2_ = counts[b]
                for i in s:
                    if i == a:
                        a_ += 1
                    elif i == b:
                        b_ += 1
                        b2_ -= 1
                    if a_ < b_ and b2_:
                        a_ = 0
                        b_ = 0
                    if a_ and b_:
                        ans = max(ans, a_ - b_)
        return ans

File: 2272-substring-with-largest-variance/2272-substring-with-largest-variance.cpp
class Solution {
public:
    int largestVariance(string s) {
        set<char> st;
        for (char c: s) st.insert(c);
        int ans = 0;
        for (char a: st) {
            for (char b: st) {
                if (a == b) continue;
                int val = 0;
                bool a_ = false, b_ = false;
                for (char i: s) {
                    if (i == a) {
                        val++;
                        a_ = true;
                    } else if (i == b) {
                        val--;
                        b_ = true;
                    }
                    if (val < 0) {
                        val = 0;
                        a_ = false;
                        b_ = false;
                    } else if (a_ && b_) {
                        ans = max(ans, val);
                    }
                }
            }
        }
        reverse(s.begin(), s.end());
        for (char a: st) {
            for (char b: st) {
                if (a == b) continue;
                int val = 0;
                bool a_ = false, b_ = false;
                for (char i: s) {
                    if (i == a) {
                        val++;
                        a_ = true;
                    } else if (i == b) {
                        val--;
                        b_ = true;
                    }
                    if (val < 0) {
                        val = 0;
                        a_ = false;
                        b_ = false;
                    } else if (a_ && b_) {
                        ans = max(ans, val);
                    }
                }
            }
        }
        return ans;
    }
};